v2/pkg/engine/resolve/errors.go

```diff
@@ -0,0 +1,109 @@
+package resolve
+
+import (
+	"fmt"
+	"strings"
+)
+
+type GraphQLError struct {
+	Message    string          `json:"message"`
+	Locations  []Location      `json:"locations,omitempty"`
+	Path       []string        `json:"path"`
+	Extensions *ErrorExtension `json:"extensions,omitempty"`
```

Comment:
Extension could contain any field

spec says next
```
GraphQL services may provide an additional entry to errors with key extensions. This entry, if set, must have a map as its value. 
```

So even code is not guaranteed to be present

Could we make an extension field a map?

Reply:
For my interface, I'm only interested in the code. Once we have the requirement to serialize more, we could change it.

Reply:
But I can change it, it makes the code a bit more verbose.